Automattic/Co-Authors-Plus
# Sidebar Plugin for Block Editor Integration

## Overview

This PR includes the addition of a panel in the Document sidebar in Gutenberg that provides the Co-Authors Plus functionality. There are a few things still to do, but I wanted to open this up now to start getting feedback. 

* A WP API endpoints to retrieve, save, and search coauthors using existing methods
* Use of the `Combobox` block editor component for the UI, including the autocomplete
* A data store to manage the Co-Authors state so we can access the data locally and globally
* A filter to enable the block editor integration, that is set to false by default
* Hide core author meta box by updating the _links entry in the response (To Do)

## Screenshots

Video of adding, removing, and re-ordering Co-Authors using the Sidebar panel.

https://user-images.githubusercontent.com/445861/118292172-59da9480-b4a6-11eb-8d45-c38387efcb34.mp4

## How to Test This PR

1. Pull down this branch for the coauthors plugin in a local WordPress development environment.
2. Create a few Guest Authors.
3. Add a new post. 
4. Observe that the Co-Authors meta box does not display in the editor. 
6. Observe the default Author selection in the "Status & Visibility" sidebar panel does not display. 
5. Observe the "Authors" panel added to the sidebar.
7. Observe the current user is assigned as the default author in the "Authors".
8. Search for the name of another author in the auto-complete section in the Authors panel and add it.
9. Do the same for a few other authors. Using the side buttons, remove and move the authors up and down. 
10. Make sure the Update button is activated when only the authors are changed and no other changes have been made.
7. Save the post and confirm the authors and their order has been saved by reloading the screen.
